src/mmrelay/migrate.py

```
@@ -0,0 +1,1317 @@
+        logger.warning("Could not read migration state file: %s", state_path)
+        return False
+
+
+def _mark_migration_completed() -> None:
+    """Mark migration as completed."""
+    state_path = _get_migration_state_path()
+    try:
+        state_path.write_text(MIGRATION_VERSION, encoding="utf-8")
+        logger.info("Migration completed and marked in: %s", state_path)
+    except (OSError, IOError) as e:
+        logger.error("Failed to mark migration as completed: %s", e)
```

Ruff (0.14.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)